### PR TITLE
FUN ADDITION: Helado Edition

### PR DESCRIPTION
**Pull request by ZomgPonies**
- Adds ice overlays to cold pressurized turfs

### CODE REVIEW

File: code/LINDA/LINDA_turf_tile.dm [hunk 1]
```diff
@@ -232,17 +238,14 @@
 	archived_cycle = air_master.current_cycle
 
 /turf/simulated/proc/update_visuals()
-	var/new_overlay_type = tile_graphic()
-	if (new_overlay_type == atmos_overlay_type)
-		return
-	var/atmos_overlay = get_atmos_overlay_by_name(atmos_overlay_type)
-	if (atmos_overlay)
-		overlays -= atmos_overlay
+	overlays = 0
```
Comment (tigercat2000) [inline code/LINDA/LINDA_turf_tile.dm]: overlays is a list, this should be either
`overlays.Cut()`
or
`overlays = list()`

File: code/LINDA/LINDA_turf_tile.dm [hunk 2]
```diff
@@ -232,17 +238,14 @@
 	archived_cycle = air_master.current_cycle
 
 /turf/simulated/proc/update_visuals()
-	var/new_overlay_type = tile_graphic()
-	if (new_overlay_type == atmos_overlay_type)
-		return
-	var/atmos_overlay = get_atmos_overlay_by_name(atmos_overlay_type)
-	if (atmos_overlay)
-		overlays -= atmos_overlay
+	overlays = list()
```
Comment (Fox-McCloud) [inline code/LINDA/LINDA_turf_tile.dm]: .Cut() would be better; this is prone to leaking.

**Comment by KasparoVy** [comment]
Wicked, nice touch.

But can you slip on the ice? :^)

**Comment by Regen1** [comment]
I think so, thats how it used to be at least.

**Comment by TheDZD** [comment]
I find this PR extremely lacking in ice puns.

**Comment by kylejen** [comment]
That's a slippery slope right there.

**Comment by Spacemanspark** [comment]
Icy a lot of puns. Chill, @TheDZD, don't need to be so cold about the lack of cool puns.

**Comment by FalseIncarnate** [comment]
Make Cryostylane cause ice overlays with a react_turf, and I definitely can support this.

**Comment by Fox-McCloud** [comment]
As Tiger pointed out, this makes update_visuals() considerably more expensive. This proc is called literally every single tick it fires---this makes it so the turfs are constantly getting rid of their overlays....which is problematic from a performance standpoint.

Also pointed out is that this would completely screw over any efforts to port TG's generic smoothing system, as that's reliant on overlays to function properly.

Can't support this if those problems are not addressed.

**Comment by ZomgPonies** [comment]
@Fox-McCloud This should resolve all concerns.

**Comment by Spacemanspark** [comment]
Generic smooting, @Fox-McCloud?

**Comment by tigercat2000** [comment]
@Spacemanspark -tg- uses a magic system for smoothing carpet/walls via overlays and icon_states

**Comment by Spacemanspark** [comment]
Oh, that.

**Comment by ZomgPonies** [comment]
Ready for review and merge, all concerns resolved.

**Comment by Plotron** [comment]
But... but I haven't seen pics!

**Comment by Fox-McCloud** [comment]
@tigercat2000 

Does this assuage all your concerns about this PR and its potential future issues/trouble it may cause (not to mention performance)?

**Comment by tigercat2000** [comment]
@Fox-McCloud it should be fine

**Comment by Plotron** [comment]
It doesn't look as nearly as good as I had hoped.

It should be more subtle and less regular!

**Comment by Regen1** [comment]
It's LINDA settings that needs to be tweaked, everything atmos related will need some tweaking before "looking good"

**Comment by Fox-McCloud** [comment]
lol. Overlays are the problem->Blames LINDA for appearance of sprite.

This has nothing to do with LINDA--it looks bad because of the sprite overlay and the layer it's on--it has absolutely nothing to do with LINDA; this overlay looked just as bad (or good, if that's your opinion) with ZAS as it did LINDA.

**Pull request merged**